frontend/src/lib/components/SaveToDashboard/saveToDashboardModalLogic.ts

```diff
@@ -0,0 +1,53 @@
+import { kea } from 'kea'
+import { dashboardsModel } from '~/models/dashboardsModel'
+import { prompt } from 'lib/logic/prompt'
+import { eventUsageLogic } from 'lib/utils/eventUsageLogic'
+
+import { saveToDashboardModalLogicType } from './saveToDashboardModalLogicType'
+
+export const saveToDashboardModalLogic = kea<saveToDashboardModalLogicType>({
+    props: {} as { fromDashboard: boolean },
+
+    actions: {
+        addNewDashboard: true,
+        setDashboardId: (id: number) => ({ id }),
+    },
+
+    reducers: {
+        _dashboardId: [null as null | number, { setDashboardId: (_, { id }) => id }],
+    },
+
+    selectors: {
+        dashboardId: [
+            (s) => [
+                s._dashboardId,
+                dashboardsModel.selectors.lastDashboardId,
+                dashboardsModel.selectors.dashboards,
+                (_, props) => props.fromDashboard,
+            ],
+            (_dashboardId, lastDashboardId, dashboards, fromDashboard) =>
+                _dashboardId || fromDashboard || lastDashboardId || (dashboards.length > 0 ? dashboards[0].id : null),
+        ],
+    },
+
+    listeners: ({ actions }) => ({
+        setDashboardId: ({ id }) => {
+            dashboardsModel.actions.setLastDashboardId(id)
+        },
+
+        addNewDashboard: async () => {
+            prompt({ key: `saveToDashboardModalLogic-new-dashboard` }).actions.prompt({
+                title: 'New dashboard',
+                placeholder: 'Please enter a name',
+                value: '',
+                error: 'You must enter name',
+                success: (name: string) => dashboardsModel.actions.addDashboard({ name }),
+            })
+        },
+
+        [dashboardsModel.actionTypes.addDashboardSuccess]: async ({ dashboard }) => {
+            eventUsageLogic.actions.reportCreatedDashboardFromModal()
+            actions.setDashboardId(dashboard!.id)
```

Comment:
GH actions says `Forbidden non-null assertion`

Reply:
This is something I'd like to fix before we reach true "typescript strict" compatibility. 

The problem is that `dashboard` is there, but `addDashboardSuccess` returns `DashboardType | null` because that's the type of the reducer.... even though `addDashboardSuccess` should always and only return `DashboardType`. 

The issue is also up here: https://github.com/keajs/kea-typegen/issues/10 ... and will be fixed eventually 😅 

I could replace `!` with `if (dashboard) { yada yada }`, but... 🤔 🤷 .